src/hooks/user/use-users-filtering.js

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { getUsers } from '../../redux/users/users.actions';
+import { config } from '../../configs';
+
+const { userRoles } = config;
+const [user, ...other] = userRoles.map((item) => item.role);
+
+const useUsersFiltering = (tabNumber) => {
+  const [filter, setFilter] = useState({});
+  const dispatch = useDispatch();
+
+  const handleTabChange = (tab) => {
+    if (!tab) {
+      setFilter((prevFilter) => ({
+        ...prevFilter,
+        roles: [user]
+      }));
+    } else {
+      setFilter((prevFilter) => ({
```

Comment:
You duplicate here your code, you can use something like this:  `roles: tab ? [...other] : [user]`

Reply:
fixed